### PR TITLE
Rework API Gateway guide to use new cross-language component

### PR DESCRIPTION
**Pull request by danielrbradley**
- Rework examples to use new package and in TS, Python & Go.
- Remove sections which relate to helper-functions or TypeScript-specific code.
- Refresh content around example (simplify, reword & reorder sections).
- Don't try to cover all permiatations, but plan to build out examples repo instead to help support.

Progress - completed to end of authorizers.

### CODE REVIEW

**Comment by danielrbradley** [comment]
```diff
diff --git a/themes/default/content/docs/guides/crosswalk/aws/api-gateway.md b/themes/default/content/docs/guides/crosswalk/aws/api-gateway.md
--- themes/default/content/docs/guides/crosswalk/aws/api-gateway.md
+++ themes/default/content/docs/guides/crosswalk/aws/api-gateway.md
@@ -274,9 +274,9 @@
 
 ### Integration Routes
 
 If neither of the above route types work for you, [AWS API Gateway _integrations_](https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-api-integration-types.html) connect an API
-Gateway endpoint to backend services that will execute code in response to requests. The previous lambda and 
+Gateway endpoint to backend services that will execute code in response to requests. The previous lambda and
 static examples use integrations internally, even if it's not evident in the simple interface exposed.
 
 Integrations give you full control over how HTTP requests are handled, and responses served, by an API Gateway
 route. If you want more flexibility than the earlier methods, to proxy HTTP requests, to integrate with

```

File: themes/default/content/docs/guides/crosswalk/aws/api-gateway.md [hunk 1]
```diff
@@ -47,242 +50,261 @@ We will see later how to assign a custom domain, SSL certificate, and/or elimina
...
+- [Lambda Function Event Handler Route](#defining-a-lambda-function-event-handler-route)
+- [Static Route Served by S3](#defining-a-static-route-served-by-s3)
+- [Integration Route](#defining-an-integration-route)
+- [OpenAPI Specification for an Entire Endpoint](#defining-an-openapi-specification-for-an-entire-endpoint)
+- [OpenAPI Specification for a Single Route](#defining-an-openapi-specification-for-a-single-route)
 
 Multiple endpoints on the same API Gateway can be defined using a combination of these techniques.
 
-### Defining a Lambda Function Event Handler Route
+### Lambda Request Handling
```
Comment (lukehoban) [inline themes/default/content/docs/guides/crosswalk/aws/api-gateway.md:61]: This title change feels less clear to me.  Any reason not to keep the previous title?

Aside - the links in the list above still use the names of the anchors from the older names.

File: themes/default/content/docs/guides/crosswalk/aws/api-gateway.md [hunk 2]
```diff
@@ -291,549 +313,452 @@ specifying the `target` property on your route.
...
-
-// Export the auto-generated AWS API Gateway base URL.
-export const url = api.url;
-```
-
-For more information AWS API Gateway Integrations, visit the
-[AWS documentation](https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-api-integration-types.html).
+{{< chooser language "typescript,python,go" / >}}
 
-### Defining an OpenAPI Specification for an Entire Endpoint
```
Comment (lukehoban) [inline themes/default/content/docs/guides/crosswalk/aws/api-gateway.md:331]: Did we end up removing this section?  It's fairly useful to be able to highlight that this is possible, even if we might want to scale back the depth of this content a little.

File: themes/default/content/docs/guides/crosswalk/aws/api-gateway.md [hunk 3]
```diff
@@ -973,23 +907,34 @@ Note that we must create this certificate in the `us-east-1` (Northern Virginia)
 ```typescript
 // Provision an SSL certificate to enable SSL -- ensuring to do so in us-east-1.
 const awsUsEast1 = new aws.Provider("usEast1", { region: "us-east-1" });
-const sslCert = new aws.acm.Certificate("sslCert", {
+const sslCert = new aws.acm.Certificate(
+  "sslCert",
+  {
```
Comment (lukehoban) [inline themes/default/content/docs/guides/crosswalk/aws/api-gateway.md:912]: Is there a reason for this re-formatting?  The original looks more like standard TS formatting.

File: themes/default/content/docs/guides/crosswalk/aws/api-gateway.md (hunk 2 above)
Comment (lukehoban) [reply themes/default/content/docs/guides/crosswalk/aws/api-gateway.md:331]: Ahh - I see it was moved to the end.

File: themes/default/content/docs/guides/crosswalk/aws/api-gateway.md [hunk 4]
```diff
@@ -291,549 +313,452 @@ specifying the `target` property on your route.
...
-const callbackRole = new aws.iam.Role("callbackRole", {
-    // Specify appropriate AWS permissions here.
-});
-
-const callbackFxn = new aws.lambda.CallbackFunction("callbackFxn", {
-    role: callbackRole,
-    callback: async (event: awsx.apigateway.AuthorizerEvent) => {
-        // --- Add custom authorization logic here. ---
-        return awsx.apigateway.authorizerResponse("user", "Allow", event.methodArn);
+const api = new awsx.apigateway.API("example", {
```
Comment (lukehoban) [inline themes/default/content/docs/guides/crosswalk/aws/api-gateway.md:719]: This and the next few still needs porting to multi-lang - right?

**Review by lukehoban — COMMENTED** [review]
A few quick notes - overall this is looking really good.  I've gone through the first half of the Go examples myself to validate, and they are looking really good.  Will go through the rest once this gets fleshed out a bit further.

File: themes/default/content/docs/guides/crosswalk/aws/api-gateway.md (hunk 1 above)
Comment (danielrbradley) [reply themes/default/content/docs/guides/crosswalk/aws/api-gateway.md:61]: I've been trying to make the title a little shorter so they don't get cut off in the side naviagation:

image

Will just drop the prefix here instead - "Lambda Function Event Handler Route" - and check all the cross-links too 👍

File: themes/default/content/docs/guides/crosswalk/aws/api-gateway.md (hunk 4 above)
Comment (danielrbradley) [reply themes/default/content/docs/guides/crosswalk/aws/api-gateway.md]: Yup - bashing through the last few sections now

File: themes/default/content/docs/guides/crosswalk/aws/api-gateway.md (hunk 3 above)
Comment (danielrbradley) [reply themes/default/content/docs/guides/crosswalk/aws/api-gateway.md:980]: No, probably an auto-formatter pluggin taking liberties.

File: themes/default/content/docs/guides/crosswalk/aws/api-gateway.md [hunk 5]
```diff
@@ -22,23 +22,26 @@ is inexpensive, has no minimum fees, and you only pay for the API calls you rece
...
 ## Overview
 
-Pulumi Crosswalk for Amazon Web Services (AWS) provides better AWS API management through significantly easier ways of programming an API Gateway. This includes using
-infrastructure as code techniques for simple, declarative APIs, including easy Lambda-based handlers.
+Pulumi Crosswalk for Amazon Web Services (AWS) provides better AWS API management through significantly easier
+ways of programming an API Gateway. This includes using infrastructure as code techniques for simple, declarative
+APIs, including easy Lambda integration.
 
-## Defining an AWS API Gateway Endpoint and Routes
+Full examples can be found in the [Pulumi Registry for the AWS API Gateway component](https://www.pulumi.com/registry/packages/aws-apigateway/).
```
Comment (susanev) [inline themes/default/content/docs/guides/crosswalk/aws/api-gateway.md:29]:
```suggestion
Full examples can be found in the [AWS API Gateway component](https://www.pulumi.com/registry/packages/aws-apigateway/) in the Pulumi Registry.
```

File: themes/default/content/docs/guides/crosswalk/aws/api-gateway.md [hunk 6]
```diff
@@ -22,23 +22,26 @@ is inexpensive, has no minimum fees, and you only pay for the API calls you rece
...
-* Adhere to the [REST](https://en.wikipedia.org/wiki/Representational_state_transfer) protocol.
-* Implement standard HTTP methods such as `GET`, `POST`, `PUT`, `PATCH`, and `DELETE`.
+- Are HTTP based.
+- Adhere to the [REST](https://en.wikipedia.org/wiki/Representational_state_transfer) protocol.
+- Implement standard HTTP methods such as `GET`, `POST`, `PUT`, `PATCH`, and `DELETE`.
 
 Each API Gateway instance defines a new API endpoint and a collection of API routes, each of which has a distinct URL.
 
 > Internally the API Gateway resource uses a collection of supporting objects, like resources, methods, and more,
 > however one of the benefits of Pulumi Crosswalk for AWS is that it hides these mechanics behind a simpler interface.
```
Comment (susanev) [inline themes/default/content/docs/guides/crosswalk/aws/api-gateway.md:43]: this isnt you, but would you mind change this to the note styles. we shouldn't be using quote styles for things that are not quotes.

```
{{% notes type="info" %}}

{{% /notes %}}
```

File: themes/default/content/docs/guides/crosswalk/aws/api-gateway.md [hunk 7]
```diff
@@ -47,242 +50,261 @@ We will see later how to assign a custom domain, SSL certificate, and/or elimina
...
-endpoint with a single API, listening at `/` for `GET` requests, which simply returns a `200 OK` for each call:
+This example creates an AWS API Gateway endpoint with a single API, listening at `/` for `GET` requests, which simply returns a `200 OK` for each call.
+
+The path can be parameterized to match specific patterns:
+
+- A literal pattern e.g. `/pets` will only match `/pets`
+- A parameterized pattern e.g. `/pets/{petId}` will match child routes such as `/pet/6sxz2j`
+- A wildcard pattern specified with `{proxy+}` e.g. `/parent/{proxy+}` will mach all decendant paths such as `/parent/child/grandchild`
+
+For more complete information about creating Lambda Functions, [see the Pulumi Crosswalk for AWS Lambda documentation]({{< relref "lambda" >}}).
```
Comment (susanev) [inline themes/default/content/docs/guides/crosswalk/aws/api-gateway.md:75]:
```suggestion
For more complete information about creating Lambda Functions, see the [Pulumi Crosswalk for AWS Lambda documentation]({{< relref "lambda" >}}).
```

File: themes/default/content/docs/guides/crosswalk/aws/api-gateway.md [hunk 8]
```diff
@@ -47,242 +50,261 @@ We will see later how to assign a custom domain, SSL certificate, and/or elimina
...
 
-### Defining an Integration Route
+If the local path points to a directory, the route will automatically be created as a proxy path (i.e. `/{proxy+}`) to match all sub-directories and the content type for all files will be inferred automatically. If the local path points to a single file you can specify the content type explicitly with the `contentType` property.
 
-If neither of the above route types work for you, AWS API Gateway uses so-called _integrations_ to hook up an API
-Gateway endpoint to backend services that will execute code in response to requests. The above examples use
-integrations internally, even if it's not evident in the simple interface exposed.
+### Integration Routes
+
+If neither of the above route types work for you, [AWS API Gateway _integrations_](https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-api-integration-types.html) connect an API
```
Comment (susanev) [inline themes/default/content/docs/guides/crosswalk/aws/api-gateway.md:305]:
```suggestion
If neither of the above route types work for you, [AWS API Gateway integrations](https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-api-integration-types.html) connect an API
```

File: themes/default/content/docs/guides/crosswalk/aws/api-gateway.md [hunk 9]
```diff
@@ -463,579 +707,808 @@ route. The validators defined at a method level override any validator set at th
...
-If you wish to reuse an Authorizer across multiple routes, you can declare it in a variable.
+{{% /choosable %}}
 
-For additional information about request-based AWS API Gateway Lambda Authorizers, see the
-[AWS documentation](https://docs.aws.amazon.com/apigateway/latest/developerguide/apigateway-use-lambda-authorizer.html).
+For additional information about request validation, refer to [Enable Request Validation in AWS API Gateway](https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-method-request-validation.html#api-gateway-request-validation-basic-definitions).
 
-#### Creating a Lambda-based Request Authorizer with an Existing Lambda
+> Request body validation is currently not supported. If you have a need for it, we would love to hear from you.
+> Comment on [this open issue](https://github.com/pulumi/pulumi-awsx/issues/198) with details about your use case.
```
Comment (susanev) [inline themes/default/content/docs/guides/crosswalk/aws/api-gateway.md:832]: can we use note styles instead? since this isn't a quote?

**Review by susanev — COMMENTED** [review]
thank you for doing this work!!

**Review by susanev — APPROVED** [review]

**Comment by lukehoban** [comment]
🎉

**Pull request merged**